cmake/SIOCreateConfig.cmake

```diff
@@ -0,0 +1,26 @@
+SET(CMAKE_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake")
```

Comment:
Why having a separate file that you include only once in the top level CMakeLists.txt?

Reply:
Mainly for a bit of separation between things that concern building and things that concern exporting things properly for usage in dependency.